File: 백준/Bronze/5919. Hay Bales/Hay Bales.py
import sys

N = int(sys.stdin.readline().strip())

piles = list()
sum = 0
for _ in range(N):
    pile = int(sys.stdin.readline().strip())
    sum += pile
    piles.append(pile)

average = sum // N

move = 0
for pile in piles:
    if pile > average:
        move += pile - average

print(move)

File: 백준/Bronze/5919. Hay Bales/Hay Bales.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        int[] piles = new int[N];
        int sum = 0;
        for (int i = 0; i < N; i++) {
            piles[i] = sc.nextInt();
            sum += piles[i];
        }
        sc.close();

        int average = sum / N;
        int move = 0;
        for (int pile : piles) {
            if (pile > average)
                move += pile - average;
        }
        System.out.println(move);
    }
}